### PR TITLE
refactor(stellar): update ITS deployment script and trusted chain

### PR DESCRIPTION
**Pull request by AttissNgo**
[AXE-6883](https://axelarnetwork.atlassian.net/browse/AXE-6883)

[AXE-6883]: https://axelarnetwork.atlassian.net/browse/AXE-6883?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: stellar/deploy-contract.js [hunk 1]
```diff
@@ -50,6 +50,14 @@ async function getInitializeArgs(config, chain, contractName, wallet, options) {
         case 'interchain_token_service': {
             const gatewayAddress = nativeToScVal(Address.fromString(chain?.contracts?.axelar_gateway?.address), { type: 'address' });
             const gasServiceAddress = nativeToScVal(Address.fromString(chain?.contracts?.axelar_gas_service?.address), { type: 'address' });
+            const hubChain = config.chains.axelar || config.axelar;
```
Comment (ahramy) [inline stellar/deploy-contract.js:53]: Just wondering, why do we need to consider both cases?

File: stellar/deploy-contract.js [hunk 2]
```diff
@@ -50,6 +50,14 @@ async function getInitializeArgs(config, chain, contractName, wallet, options) {
...
+            const itsHub = hubChain.contracts.InterchainTokenService.address;
```
Comment (ahramy) [inline stellar/deploy-contract.js:54]: You could use `validateParameters` for validation check. Reference: https://github.com/axelarnetwork/axelar-contract-deployments/blob/04a0dbd71cf2cda3e4791e8363f364379ae53830/stellar/deploy-contract.js#L148

File: stellar/deploy-contract.js (hunk 1 above)
Comment (AttissNgo) [reply stellar/deploy-contract.js:53]: We don't. This is a mistake

File: stellar/its.js [hunk 3]
```diff
@@ -5,21 +5,20 @@ const { saveConfig, loadConfig, addOptionsToCommands, getChainConfig } = require
...
 
-async function removeTrustedAddress(wallet, _, chain, contractConfig, arg, options) {
+async function removeTrustedChain(wallet, _, chain, contractConfig, arg, options) {
     const contract = new Contract(contractConfig.address);
     const callArg = nativeToScVal(arg, { type: 'string' });
 
-    const operation = contract.call('remove_trusted_address', callArg);
+    const operation = contract.call('remove_trusted_chain', callArg);
 
     await broadcast(operation, wallet, chain, 'Trusted Address Removed', options);
```
Comment (milapsheth) [inline stellar/its.js:23]:
```suggestion
    await broadcast(operation, wallet, chain, 'Trusted Chain Removed', options);
```

File: stellar/README.md [hunk 4]
```diff
@@ -85,8 +85,16 @@ node stellar/deploy-contract.js deploy axelar_operators --chain-name <CHAIN_NAME
...
+### Interchain Token
+
+```bash
+node stellar/deploy-contract.js deploy interchain_token --chain-name <CHAIN_NAME> --wasm-path ../axelar-cgp-soroban/target/wasm32-unknown-unknown/release/interchain_token.optimized.wasm
+```
+
 ### Interchain Token Service
 
+_Note: The Interchain Token Service constructor requires the wasm hash of the Interchain Token. Please ensure Interchain Token is deployed before deploying Interchain Token Service._
+
```
Comment (milapsheth) [inline stellar/README.md:97]:
```suggestion
### Interchain Token Service

Deploy Interchain Token wasm first.
```bash
node stellar/deploy-contract.js deploy interchain_token --chain-name <CHAIN_NAME> --wasm-path ../axelar-cgp-soroban/target/wasm32-unknown-unknown/release/interchain_token.optimized.wasm
```

Deploy and initialize ITS contract next.

```

**Review by milapsheth — APPROVED** [review]

**Pull request merged**